# Assignment 4.1

## Measuring shifts between images

- Use imexam() to measure the coordinates of at least five stars in the two images science1V.FIT and science10V.FIT. Save the coordinates of the stars in a text file for each image (e.g. coo1V.txt and coo10V.txt) with the x and y coordinates in column 0 and 1, respectively.

You can just copy-and-paste the output from imexam() into a text editor and edit as necessary to keep the numerical values only. Make sure the stars appear in the same order in both files. You will need these files later!

## Hand in:

- Your lists of the coordinates of the stars in each image (40/100 pts)
- The calculated mean shifts between the two images (30/100 pts)
- How well do the shifts based on the individual stars agree? Be as quantitative as possible! (30/100 pts)

You can fill in the answers in the Markdown field here below and add your Python commands at the end of the notebook.
When you are done, download the notebook as a PDF file, and submit via Brightspace. 

### Coordinates in science1V.FIT:
1047.4767	721.1090
1083.0567	727.0286
1111.3292	743.3196
1098.0349	765.4710
1123.2623	766.8744
### Coordinates in science10V.FIT:
1052.6754	715.3699
1088.2183	721.2954
1116.5359	737.5746
1103.2338	759.7435
1128.4889	761.1085


Here below you can insert your Python commands (feel free to add cells as needed)

In [1]:
import astropy.io.fits as fits
import numpy as np
import imexam
import os
import time

Ginga not installed, use other viewer, or no viewer


In [3]:
def main(filepath):
    viewer = imexam.connect()
    time.sleep(5)
    viewer.load_fits(filepath)
    viewer.zoomtofit()
    viewer.imexam()
    
if __name__ == "__main__":
    filepath = "../science/science10V.FIT"
    main(filepath)

XpaException: Unknown XPA Error : XPAGet returned 0!

science1V.FIT
1047.4767	721.1090
1083.0567	727.0286
1111.3292	743.3196
1098.0349	765.4710
1123.2623	766.8744

science10V.FIT
1052.6754	715.3699
1088.2183	721.2954
1116.5359	737.5746
1103.2338	759.7435
1128.4889	761.1085

In [4]:
coordinates1 = np.array([
    [1047.4767,721.1090],
    [1083.0567,727.0286],
    [1111.3292,743.3196],
    [1098.0349,765.4710],
    [1123.2623,766.8744]])

coordinates10 = np.array([
    [1052.6754,715.3699],
    [1088.2183,721.2954],
    [1116.5359,737.5746],
    [1103.2338,759.7435],
    [1128.4889,761.1085]])

shifts = coordinates1-coordinates10
mean_shift = np.mean(shifts, axis=0)
std = np.std(shifts, axis=0)
distances = np.linalg.norm(shifts, axis=1)
std_distances = np.std(distances)

print("Mean shift: {}".format(mean_shift))
print("Standard deviation (for both axes separately): {}".format(std))
print("Standard deviation (of the distance between the two points): {}".format(std_distances))

Mean shift: [-5.1985   5.74214]
Standard deviation (for both axes separately): [0.02106495 0.01323792]
Standard deviation (of the distance between the two points): 0.022309453982365512


Mean shift: [-5.1985   5.74214]

Standard deviation (for both axes separately): [0.02106495 0.01323792]

Standard deviation (of the distance between the two points): 0.022309453982365512